bcdc_apitests/tests/resources/test_resources.py

```diff
@@ -23,87 +23,141 @@ def test_resource_create(ckan_url, ckan_rest_dir, ckan_auth_header,
 
-    try:
-        res_data = remote_api.action.resource_update(**resource_data)
-        logger.debug("resource_update: %s", res_data)
-        # TODO: KN, put in an assertion here that verifies that you successfully updated
+        # update resource data dic with package id
+        resource_data['package_id'] = res_data['results'][0]['package_id']
+        logger.debug("resource_pkgID: %s", resource_data['package_id'])
+
+        # add id key to resource data dic
+        resource_data['id'] = res_data['results'][0]['id']
+        logger.debug("resource_id: %s", resource_data['id'])
+
+        # change some data in resource data dic
+        resource_data['description'] = 'test resource description changed'
+
+        #define api
+        api_call = '{0}{1}/{2}'.format(ckan_url, ckan_rest_dir, 'resource_update')
+        logger.debug('api_call: %s', api_call)
+
+        #update resource
+        res_data = requests.post(api_call, headers=ckan_auth_header, json=resource_data)
+        logger.debug("resource_update: %s", res_data.text)
+
+        assert res_data.status_code == 200
 
     except ckanapi.CKANAPIError as err:
         logger.debug("err: %s %s", type(err), err)
 
 
 # search resource
 def test_resource_search(remote_api_admin_auth, resource_data):
-
+    '''
+    :param remote_api_admin_auth: ckanapi remote, with auth
+    :param resource_data: test resource structure
+    '''
     try:
+        # define remote api
         remote_api = remote_api_admin_auth
         logger.debug("resource_name: %s", resource_data['name'])
-        # using search to get the ID of resource.  needs to migrate as fixture via package query
-        # TODO: For some reason this isn't working?
+
+        # search for resource by name
         res_data = remote_api.action.resource_search(query="name:{0}".format(resource_data['name']))
         logger.debug("resource search: %s", res_data)
 
-        # res_id = res_data['results'][0]['id']
-        logger.debug("resource_name: %s", resource_data['name'])
-        # TODO: once the search gets fixed uncomment
-        #assert res_data['count'] >= 1
+        assert res_data['count'] >= 1
+
     except ckanapi.CKANAPIError as err:
         logger.debug("err: %s %s", type(err), err)
 
 
 # delete resource
-def test_resource_delete(remote_api_admin_auth, resource_data):
+def test_resource_delete(remote_api_admin_auth,ckan_url,ckan_rest_dir,ckan_auth_header, resource_data):
     '''
     :param remote_api_admin_auth: ckanapi remote, with auth
+    :param resource_data: test resource structure
     '''
     try:
-
+        # define api
         remote_api = remote_api_admin_auth
 
-        # res_data = remote_api.action.resource_search(query="name:test resource")
-        # resId = res_data['results'][0]['id']
-        # logger.debug("resource_id: %s", resId)
+        # search to get the ID of resource.
+        res_data = remote_api.action.resource_search(query="name:{0}".format(resource_data['name']))
```

Comment:
The resource name is going to be consistently defined in the config_fixture.py file.  Thinking you could move the resource_search code to that fixture and have the fixture return the id.  That will reduce duplication.